warden/x/warden/keeper/msg_server_remove_keychain_admin.go

```
@@ -0,0 +1,44 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	types "github.com/warden-protocol/wardenprotocol/warden/x/warden/types/v1beta3"
+)
+
+func (k msgServer) RemoveKeychainAdmin(goCtx context.Context, msg *types.MsgRemoveKeychainAdminRequest) (*types.MsgRemoveKeychainAdminResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	kr, err := k.keychains.Get(ctx, msg.KeychainId)
+	if err != nil {
+		return nil, err
+	}
+
+	if !kr.IsAdmin(msg.Authority) {
+		return nil, fmt.Errorf("keychain updates should be requested by admins")
+	}
+	if !kr.IsAdmin(msg.Admin) {
+		return &types.MsgRemoveKeychainAdminResponse{}, nil
```

as above, let's return an error here